index.js

```diff
@@ -237,11 +240,11 @@ class KeyringController extends EventEmitter {
    * @returns {Promise<Keyring>} The new keyring.
    */
   async addNewKeyring(type, opts) {
-    const Keyring = this.getKeyringClassForType(type);
-    const keyring = new Keyring(opts);
+    const keyring = await this._newKeyring(type, opts);
+
     if ((!opts || !opts.mnemonic) && type === KEYRINGS_TYPE_MAP.HD_KEYRING) {
       keyring.generateRandomMnemonic();
-      keyring.addAccounts();
+      await keyring.addAccounts();
```

Comment:
This method is asynchronous. I've taken the liberty to add an `await`.

Reply:
Nit: could we move this to a separate PR? We could merge this in isolation, since it's not directly tied to the other changes. That would make it easier to understand the commit history and prepare the changelog and so on.

Reply:
It would also be great to have a test for this, to demonstrate that when `addAccounts` is doing asynchronous work, this function guarantees that the accounts have been added by the time it resolves.

Reply:
Yes, better on a different PR. I've removed it from here and will open another one.